### ISSUE TITLE
Simultaneous (bulk) testport for all combinations of port options

### ISSUE THREAD

**Issue by woodsb02** [freebsd/poudriere#303]
Please consider adding an option to perform simultaneous (bulk) testport runs for all combinations of port options. This would be useful for two purposes:
1. Confirming port builds successfully for all combinations of options
2. Save the output of `make makeplist` for each combination of options, and merge them all once finished to automatically generate a pkg-plist with %%OPT%% prefixes on lines that differ as options vary.

Note: The standard "/you/have/to/check/what/makeplist/gives/you" would remain as the first line in the automatically generated pkg-plist.

**Comment by bdrewery** [comment]
Just in my `security/openssh-port` with 10 options there are 1024 unique combinations. https://www.wolframalpha.com/input/?i=combinations+in+10+items

**Comment by woodsb02** [comment]
That is true - so how is a port maintainer supposed to test every combination works?
Most of the time it is good enough to try with all options unset, each option set on its own, and then with all options set (just to double check if having multiple options set changes pkg-plist). Perhaps this could do that?
Lastly, building 1024 ports simultaneously isn't that big a deal, considering the ports tree has far more than that and many people regularly build the entire tree with poudriere bulk.

**Comment by bdrewery** [comment]
#601 will be needed for this first. I think it's a good idea to at least allow a user to specify their own set of options sets to test concurrently, rather than auto determining it.

**Comment by mat813** [comment]
In the mean time, you can (ab)use flavors (if the port is not flavored, of course).
For the BIND9 ports, I have an ugly Makefile.local file containing:

```
FLAVORS=        normal small huge mysql ${OPTIONS_SINGLE_GSSAPI:NGSSAPI_NONE:tl}
.if empty(FLAVOR)
FLAVOR= normal
.endif

.if ${FLAVOR} != normal
PKGNAMESUFFIX:= ${PKGNAMESUFFIX}-${FLAVOR}
.endif

.if ${FLAVOR} == mysql
OPTIONS_SET_FORCE=      DLZ_MYSQL
OPTIONS_UNSET_FORCE=    THREADS DNSTAP
.elif ${FLAVOR} == small
OPTIONS_UNSET_FORCE=    SSL ${OPTIONS_DEFINE} ${OPTIONS_GROUP_DLZ}
.elif ${FLAVOR} == huge
OPTIONS_SET_FORCE=      ${OPTIONS_DEFINE} ${OPTIONS_GROUP_DLZ:NDLZ_MYSQL}
.elif ${FLAVOR:Mgssapi*}
OPTIONS_UNSET_FORCE=    GSSAPI_NONE
OPTIONS_SET_FORCE=      ${FLAVOR:tu}
.endif

CONFIGURE_ARGS+=        --enable-full-report
```

Which kinda tries all the possibilities I am interested in.

**Comment by lbartoletti** [comment]
I am interested too by this. I often use makeplist from `ports-mgmt/bsdadminscripts2`but sometime it fails (at least on one difficult port)

**Comment by Zirias** [comment]
Ok, regarding #870, I didn't find *this* issue, sorry. Shouldn't it have a feature-request label?

Anyways, while building all in parallel and supporting `pkg-plist` creation would be *awesome*, I think just automating `testport` with all valid combinations of options would already be a huge plus. Kind of like the minimal valuable version of this request. Is this still unfeasible?
